test/test_torch.py

```diff
@@ -16656,6 +16656,39 @@ def test_dot(self, device, dtype):
         torch.dot(v1, v2, out=out)
         self.assertEqual(res1, out)
 
+    @dtypes(torch.float, torch.double, torch.cfloat, torch.cdouble)
+    def test_vdot(self, device, dtype):
```

Comment:
Would you also add tests for:

-  dot and vdot getting arguments of different dtypes, incorrect number of dimensions, and mismatched devices (this test is a little weird to write in the device generic framework, but you can check that your current device is a cuda device and then create a tensor on that device + a cpu tensor to do it)
- out behavior, including mismatched out dtype
- after the method variant is added, you can add a test here, too:

https://github.com/pytorch/pytorch/blob/eb47940c0a773ec52e67778dbe01918f6b162271/test/test_torch.py#L19786

Reply:
https://github.com/pytorch/pytorch/pull/43004/files#r470634289